script.js

```diff
@@ -1,149 +1,283 @@
+    <div class="movie-directors">
+      <h5>${movie.director}</h5>
+      <h5>${movie.movie_year}</h5>
     </div>
-    <div class="star-output" id="output">
-      <h5></h5>
-    </div>  
-  </div>
-</div>
 `;
 
-const headerMarkup = `
-<div class="about-movie hidden" id="about-movie">
-  <div>
+const headerMarkup = movie => `
+<div class="about-movie hidden">
   <div class="movie-title">
     <h3>${movie.title}</h3>
     <h5>${movie.genre}</h5>
-  </div>
-
-  <div class="buy-movie">
-    <h5 class="buy-movie__price">${movie.price}</h5>
-    <a href="#" class="buy-movie__btn">Buy</a>
-  </div>
-  </div>
   <div class="movie-description">
     <p>
-      actors: ${movie.actors}
+       ${movie.actors.slice(0, 2).join(' , ')} , . . .
     </p>
     <p>
-      ${movie.description}
+      ${movie.description.split(' ').slice(0, 22).join(' ')} . . .
      </p>
   </div>
-     <div class="movie-photo"></div>  
+    <div class="buy-movie">
+      <h5 class="buy-movie__price">$${movie.price} </h5>
+      <a href="#" class="buy-movie__btn">Buy</a>
+    </div>
+  </div>
+     <div class="movie-trailer" id="stream">
+     <iframe src=${
+       movie.trailer
+     } title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe>
+     </div>
 </div>
   `;
 
+const cardBarMarkup = movie => `
+      <p>rate this movie</p>
+      <div class="card-bar1"></div>
+      <div class="card-bar2"></div>
+      <div class="card-bar3"></div>`;
+
+let cards;
+
+/* --- function to create new element */
 function createElements(parent, elType, myClass) {
   const element = document.createElement(elType);
   element.classList.add(myClass);
   parent.appendChild(element);
+
   return element;
 }
-// design and implement the layout of the movie card via DOM manipulation{}
-function showMovieCard() {
-  const cardElement = createElements(main, 'div', 'card');
-
-  cardElement.insertAdjacentHTML('afterbegin', cardMarkup);
-  //  a. submitting and displaying a comment about the movie. Implement an imput under the movie and a functionality to display the submitted comment.
-  const clientMovieRate = document.querySelector('.movie-rating');
-  const inputElement = document.getElementById('inputValue');
-  const commentBtn = document.querySelector('.comment-btn');
-  const commentOutput = document.querySelector('.comment-output');
-
-  const stars = document.querySelectorAll('.star');
-  const starsOutput = document.getElementById('output');
-
-  cardElement.addEventListener('mouseover', function (e) {
-    clientMovieRate.classList.remove('hidden');
-  });
 
-  cardElement.addEventListener('mouseleave', function () {
-    clientMovieRate.classList.add('hidden');
+async function getMovies() {
+  const response = await fetch(apiUrl);
+  const data = await response.json();
+
+  displayMovie(data);
+}
```

Comment:
> A lot of users reported issues with the page not working and just being blank. After a long investigation you find that GitHub was was undergoing maintenance and was thus unavailable for 2 whole days. Though it is not your fault you still want to handle such issues more gracefully in the future. You decide that it would improve that user experience showing an error message when the API is unavailable or slow. How can you do that?